tools/src/main/java/org/apache/kafka/trogdor/coordinator/NodeManager.java

```diff
@@ -203,40 +205,31 @@ public void run() {
                 }
-                // Identify tasks which are running, but which we don't know about.
-                // Add these to the NodeManager as tasks that should not be running.
                 for (Map.Entry<String, WorkerState> entry : agentStatus.workers().entrySet()) {
                     String id = entry.getKey();
                     WorkerState state = entry.getValue();
-                    if (!workers.containsKey(id)) {
+                    ManagedWorker worker = workers.get(id);
+                    if (worker == null) {
+                        // Identify tasks which are running, but which we don't know about.
+                        // Add these to the NodeManager as tasks that should not be running.
                         log.warn("{}: scheduling unknown worker {} for stopping.", node.name(), id);
                         workers.put(id, new ManagedWorker(id, state.spec(), false, state));
-                    }
-                }
-                // Handle workers which need to be stopped.  Handle workers which have newly completed.
-                for (Map.Entry<String, WorkerState> entry : agentStatus.workers().entrySet()) {
-                    String id = entry.getKey();
-                    WorkerState state = entry.getValue();
-                    ManagedWorker worker = workers.get(id);
-                    if (state instanceof WorkerStarting || state instanceof WorkerRunning) {
-                        if (!worker.shouldRun) {
-                            worker.tryStop();
-                        }
-                    } else if (state instanceof WorkerDone) {
-                        if (!(worker.state instanceof WorkerDone)) {
-                            WorkerDone workerDoneState =  (WorkerDone) state;
-                            String error = workerDoneState.error();
-                            if (error.isEmpty()) {
-                                log.info("{}: Worker {} finished with status '{}'",
-                                    node.name(), id, workerDoneState.status());
-                            } else {
-                                log.warn("{}: Worker {} finished with error '{}' and status '{}'",
-                                    node.name(), id, error, workerDoneState.status());
+                    } else {
+                        // Handle workers which need to be stopped.
+                        if (state instanceof WorkerStarting || state instanceof WorkerRunning) {
+                            if (!worker.shouldRun) {
+                                worker.tryStop();
                             }
-                            taskManager.handleWorkerCompletion(node.name(), worker.id, error);
+                        }
+                        // Notify the TaskManager if the worker state has changed.
+                        if (!worker.state.equals(state)) {
+                            log.info("{}: WATEREMLON: worker state changed to {}", node.name(), state);
```

Comment:
what's " WATEREMLON"?

Reply:
Sorry, I forgot to take that out.  Let me fix up these logs messages

Reply:
@cmccabe Looks like the log entry hasn't been updated.

Reply:
fixed.  I wasn't able to search for it because it was misspelled :disappointed: